web/src/app/components/map/map.component.ts

```diff
@@ -307,15 +319,21 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     this.changeDetectorRef.detectChanges();
   }
 
-  private onMarkerDragEnd(event: google.maps.MouseEvent, loi: PointOfInterest) {
+  private onMarkerDragEnd(
+    event: google.maps.MouseEvent,
+    id: string,
+    jobId: string
+  ) {
     this.newLatLng = new google.maps.LatLng(
       event.latLng.lat(),
       event.latLng.lng()
     );
-    this.newLocationOfInterestToReposition = new PointOfInterest(
-      loi.id,
-      loi.jobId,
-      new firebase.firestore.GeoPoint(event.latLng.lat(), event.latLng.lng())
+
+    this.newLocationOfInterestToReposition = new GenericLocationOfInterest(
+      id,
+      jobId,
+      new Point(new Coordinate(event.latLng.lat(), event.latLng.lng())),
+      ImmutableMap<string, string | number>()
```

Comment:
nit: is it possible to unify how we construct this field? There is `ImmutableMap<string, string | number>()`, `ImmutableMap()` `Map<string, string | number>()` & `Map()` in this PR. Maybe stick to `Map()` since it's the easiest?